contracts/adapters/dp/MakerVault.sol

```diff
@@ -361,4 +367,114 @@ contract MakerVaultAdapter is IDebtPosition, FactoryFriendly {
 
         return result;
     }
+
+    /// @notice Returns the current collateralization ratio of the vault.
+    /// @return ratio as fixed point ray (27 decimals)
+    function ratio() public view override returns (uint256) {
+        // Collateralization Ratio = Vat.urn.ink * Vat.ilk.spot * Spot.ilk.mat / (Vat.urn.art * Vat.ilk.rate)
+        // or
+        // Collateralization Ratio = collateral in vault * spot price * liquidation ratio / (dait debt)
+        // or
+        // r = (c * s * l) / d
+
+        // wad - 18 decimal places
+        // ray - 27 decimal places
+        // rad - 45 decimal places
+        // struct Vat.Ilk {
+        //     uint256 Art;   // Total Normalised Debt     [wad]
+        //     uint256 rate;  // Accumulated Rates         [ray]
+        //     uint256 spot;  // Price with Safety Margin  [ray]
+        //     uint256 line;  // Debt Ceiling              [rad]
+        //     uint256 dust;  // Urn Debt Floor            [rad]
+        // }
+        // struct Vat.Urn {
+        //     uint256 ink;   // Locked Collateral  [wad]
+        //     uint256 art;   // Normalised Debt    [wad]
+        // }
+        // struct Spot.Ilk {
+        //     PipLike pip;  // Price Feed
+        //     uint256 mat;  // Liquidation ratio [ray]
+        // }
+
+        return _div(collateralValue(), debtValue());
+    }
+
+    /// @notice returns the amount of asset that should be repaid to
+    /// bring vault to target ratio.
+    /// @return amount fixed point scaled to asset decimals
+    function delta() external view override returns (uint256) {
+        uint256 debt = debtValue();
+
+        // r = (c * s * l) / d
+        uint256 debtTarget = _div(collateralValue(), ratioTarget);
+
+        // from ray to ERC20.decimals
+        uint256 scaleDownFactor = 27 - IERC20(asset).decimals();
```

Comment:
What happens if some crazy asset has more than 27 decimals?

Reply:
I was figuring this will be always be DAI since that is the currently the only maker debt coin? If that's the case, then maybe it does not make sense to generalize this line

Reply:
I think this is a pretty safe assumption given that `wad` is a pretty central precision level in maker. Like if you tried to use a token with some other level of precision, there's a good chance it would break math elsewhere in the system.